### PR TITLE
fix property predicates

### PR DESCRIPTION
**Pull request by nanoMichal**
PropertyIsClass and PropertyIsInstance were not working properly.

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/ruler/predicate/PropertyIsClass.js [hunk 1]
```diff
@@ -40,11 +41,11 @@ foam.CLASS({
 
       if ( getIsNew() ) {
         FObject nu  = (FObject) NEW_OBJ.f(obj);
-        return CLASS_OF(getOf().getObjClass()).f(nu.getClassInfo().getAxiomByName(getPropName()));
+         return (new IsClassOf(getOf())).f(nu.getProperty(getPropName()));
```
Comment (kgrgreer) [inline src/foam/nanos/ruler/predicate/PropertyIsClass.js]: Couldn't this just be getOf().isInstance(nu[getPropName()]); ?

File: src/foam/nanos/ruler/predicate/PropertyIsInstance.js [hunk 2]
```diff
@@ -40,11 +40,11 @@ foam.CLASS({
 
       if ( getIsNew() ) {
         FObject nu  = (FObject) NEW_OBJ.f(obj);
-        return INSTANCE_OF(getOf().getObjClass()).f(nu.getClassInfo().getAxiomByName(getPropName()));
+        return getOf().isInstance(nu.getProperty(getPropName()));
```
Comment (kgrgreer) [inline src/foam/nanos/ruler/predicate/PropertyIsInstance.js]: Is this line correct?

File: src/foam/nanos/ruler/predicate/PropertyIsClass.js (hunk 1 above)
Comment (kgrgreer) [inline src/foam/nanos/ruler/predicate/PropertyIsClass.js:44]: getOf().getObjClass() == nu.getProperty(getPropName()).getClass(); OR
getOf().isInstance(nu.getProperty(getPropName()));

The first line is only exact match, the second handles sub-classes.

Comment (kgrgreer) [reply src/foam/nanos/ruler/predicate/PropertyIsClass.js:44]: My mistake. Carry on :(.

**Pull request merged**